charts/orchestrator/templates/rhdh-operator.yaml

```diff
@@ -44,23 +44,20 @@ metadata:
   name: backstage-backend-auth-secret
   namespace: {{ .Values.rhdhOperator.subscription.namespace }}
 stringData:
-  BACKEND_SECRET: {{ randAlphaNum 24 | nospace | b64enc }}
-{{- if ne .Values.rhdhOperator.k8s.clusterToken "" }}
+{{- $old_secret := lookup "v1" "Secret" .Values.rhdhOperator.subscription.namespace "backstage-backend-auth-secret" }}
+{{- $generated_secret := randAlphaNum 24 | nospace | b64enc }}
```

Comment:
we can remove the `| b64enc`, right? it will be encoded when the script is applied.
(I don't assume double encoding will cause any harm, but it isn't needed AFAICT)

Reply:
Unfortunately no: `stringData` will encode the value of the field, so it needs to be unencoded. But if we recover the value from the existing secret, we must ensure that it is decoded before it is reprocessed by the API server and encoded once again:
new secret -> encode it -> decode it to store it in stringData -> encode it in the API server.
old encoded secret -----> decode it to store it in stringData -> encode it in the API server.

Reply:
Otherwise it will keep updating it on each reconciliation since it detects the value is different. 
new secret -> ~encode it -> decode it to store it in stringData~ -> encode it in the API server.
old encoded secret ~-----> decode it to store it in stringData~ -> encode it in the API server (double encoding).